spring-session/src/main/java/org/springframework/session/jdbc/JdbcOperationsSessionRepository.java

```diff
@@ -0,0 +1,483 @@
+	}
+
+	/**
+	 * Create a new {@link JdbcOperationsSessionRepository} instance which uses the
+	 * provided ${JdbcOperations} to manage sessions.
+	 * @param jdbcOperations the {@link JdbcOperations} to use
+	 */
+	public JdbcOperationsSessionRepository(JdbcOperations jdbcOperations) {
+		Assert.notNull(jdbcOperations, "JdbcOperations must not be null");
+		this.jdbcOperations = jdbcOperations;
+	}
+
+	@Override
+	public void setApplicationEventPublisher(ApplicationEventPublisher applicationEventPublisher) {
+		Assert.notNull(applicationEventPublisher, "ApplicationEventPublisher must not be null");
+		this.applicationEventPublisher = applicationEventPublisher;
+	}
+
+	/**
+	 * Set the prefix for database table used to store sessions.
+	 * @param tablePrefix the table prefix
+	 */
+	public void setTablePrefix(String tablePrefix) {
+		this.tablePrefix = tablePrefix;
+	}
+
+	/**
+	 * Set the maximum inactive interval in seconds between requests before newly created
+	 * sessions will be invalidated. A negative time indicates that the session will never
+	 * timeout. The default is 1800 (30 minutes).
+	 * @param defaultMaxInactiveInterval the maximum inactive interval in seconds
+	 */
+	public void setDefaultMaxInactiveInterval(Integer defaultMaxInactiveInterval) {
+		this.defaultMaxInactiveInterval = defaultMaxInactiveInterval;
+	}
+
+	public void setLobHandler(LobHandler lobHandler) {
+		Assert.notNull(lobHandler, "LobHandler must not be null");
+		this.lobHandler = lobHandler;
+	}
+
+	@SuppressWarnings("unchecked")
+	public void setSerializer(Serializer<MapSession> serializer) {
+		Assert.notNull(serializer, "Serializer must not be null");
+		this.serializer = new SerializingConverter((Serializer) serializer);
+	}
+
+	@SuppressWarnings("unchecked")
+	public void setDeserializer(Deserializer<MapSession> deserializer) {
+		Assert.notNull(deserializer, "Deserializer must not be null");
+		this.deserializer = new DeserializingConverter((Deserializer) deserializer);
+	}
+
+	@Override
+	public JdbcSession createSession() {
+		JdbcSession session = new JdbcSession();
+		if (this.defaultMaxInactiveInterval != null) {
+			session.setMaxInactiveIntervalInSeconds(this.defaultMaxInactiveInterval);
+		}
+		return session;
+	}
+
+	@Override
+	public void save(final JdbcSession session) {
+		final byte[] sessionBytes = this.serializer.convert(session.delegate);
+
+		if (session.isNew()) {
+			this.jdbcOperations.update(getQuery(Query.CREATE_SESSION), new PreparedStatementSetter() {
+
+				@Override
+				public void setValues(PreparedStatement ps) throws SQLException {
+					ps.setString(1, session.getId());
+					ps.setLong(2, session.getLastAccessedTime());
+					ps.setString(3, session.getPrincipalName());
+					JdbcOperationsSessionRepository.this.lobHandler.getLobCreator()
+							.setBlobAsBytes(ps, 4, sessionBytes);
+				}
+
+			});
+			session.setNew(false);
+			publishEvent(new SessionCreatedEvent(this, session));
```

Comment:
This won't be available to all JVM instances which is inconsistent with the way Spring Session works. We should probably remove this.